src/modules/jwt-encryption-service/JwtEncryptionService.ts

```diff
@@ -0,0 +1,49 @@
+import 'reflect-metadata';
+
+import { randomBytes } from 'crypto';
+import { JWE, JWK } from 'node-jose';
+
+import CompanyAuthConfig from 'app/models/CompanyAuthConfig';
+
+interface AuthPayload {
+    nonce: string;
+    content: string;
+}
+
+export default class JwtEncryptionService {
+    public constructor(private companyAuthConfig: CompanyAuthConfig) {}
```

Comment:
Config has 5 properties while only one is used, don't you think it is bad design? I would think it is better to keep that class generic as JWE encryption service which takes payload and key and returns encrypted payload.